src/components/CreateVirtualFacilityModal.vue

```diff
@@ -0,0 +1,195 @@
+        </ion-button>
+      </ion-buttons>
+      <ion-title>{{ translate("New parking") }}</ion-title>
+    </ion-toolbar>
+  </ion-header>
+
+  <ion-content>
+    <form @keyup.enter="createVirtualFacility">
+      <ion-list>
+        <ion-item>
+          <ion-label position="floating">
+            {{ translate("Name") }} <ion-text color="danger">*</ion-text>
+          </ion-label>
+          <ion-input @ionBlur="setFacilityId($event)" v-model="formData.facilityName"/>
+        </ion-item>
+        <ion-item ref="facilityId">
+          <ion-label position="floating">
+            {{ translate("Internal ID") }}
+          </ion-label>
+          <ion-input v-model="formData.facilityId" @keyup="validateFacilityId" @ionBlur="markFacilityIdTouched" />
+          <ion-note slot="error">
+            {{ translate('Internal ID cannot be more than 20 characters.') }}
+          </ion-note>
+        </ion-item>
+        <ion-item>
+          <ion-label position="floating">
+            {{ translate("Description") }}
+          </ion-label>
+          <ion-input v-model="formData.description"/>
+        </ion-item>
+      </ion-list>
+
+      <ion-fab vertical="bottom" horizontal="end" slot="fixed">
+        <ion-fab-button @click="createVirtualFacility" @keyup.enter.stop>
+          <ion-icon :icon="addOutline" />
+        </ion-fab-button>
+      </ion-fab>
+    </form>
+  </ion-content>
+</template>
+
+<script lang="ts">
+import {
+  IonButton,
+  IonButtons,
+  IonContent,
+  IonFab,
+  IonFabButton,
+  IonHeader,
+  IonIcon,
+  IonInput,
+  IonItem,
+  IonLabel,
+  IonList,
+  IonNote,
+  IonText,
+  IonTitle,
+  IonToolbar,
+  modalController
+} from "@ionic/vue";
+import { defineComponent } from "vue";
+import { addOutline, closeOutline } from "ionicons/icons";
+import { translate } from '@hotwax/dxp-components'
+import { FacilityService } from "@/services/FacilityService";
+import { mapGetters, useStore } from 'vuex'
+import { hasError } from "@/adapter";
+import { showToast } from "@/utils";
+import logger from "@/logger";
+
+export default defineComponent({
+  name: "CreateVirtualFacility",
+  components: {
+    IonButton,
+    IonButtons,
+    IonContent,
+    IonFab,
+    IonFabButton,
+    IonHeader,
+    IonIcon,
+    IonInput,
+    IonItem,
+    IonLabel,
+    IonList,
+    IonNote,
+    IonText,
+    IonTitle,
+    IonToolbar
+  },
+  computed: {
+    ...mapGetters({
+      virtualFacilities: 'facility/getVirtualFacilities',
+    })
+  },
+  data() {
+    return {
+      formData: {
+        facilityName: '',
+        facilityId: '',
+        description: '',
+      },
+    }
+  },
+  ionViewWillEnter() {
+    this.clearFormData()
+  },
+  methods: {
+    clearFormData() {
+      this.formData = {
+        facilityName: '',
+        facilityId: '',
+        description: '',
+      }
+    },
+    setFacilityId(event: any) {
+      this.formData.facilityId = event.target.value.trimEnd().trimStart().toUpperCase().split(' ').join('_');
```

Comment:
Why we have not used trim() here instead of using separate method trimEnd and trimStart?